### ISSUE TITLE
spacewalk-reports providing no output since 2022.5

### ISSUE THREAD

**Issue by digdilem** [uyuni-project/uyuni#5458]
Presumably linked to the changes mentioned in the [release notes](https://www.uyuni-project.org/doc/2022.05/release-notes-uyuni-server.html#_spacewalk_report_now_uses_data_from_the_reporting_database)

# Issue
I run several reports using spacewalk-report which are ingested by a script elsewhere. 

Output since this upgrade to 2022.5, output is empty, apart from field names.

It does not seem to matter what report I generate, I just get the first line and no content. 

# Example:
```
uyuni01:~ # spacewalk-report channels
mgm_id,channel_id,channel_label,channel_name,number_of_packages,organization,synced_date
uyuni01:~ # spacewalk-report users
mgm_id,organization,user_id,username,last_name,first_name,position,email,role,creation_time,last_login_time,active,synced_date
uyuni01:~ # spacewalk-report inventory
mgm_id,system_id,profile_name,hostname,ip_address,ipv6_address,registered_by,registration_time,last_checkin_time,kernel_version,packages_out_of_date,errata_out_of_date,software_channel,configuration_channel,entitlements,system_group,organization,virtual_host,architecture,is_virtualized,virt_type,hardware,minion_id,machine_id,synced_date
```

Using 2022.4 and previous, I would get the data:
```
uyuni01:~ # spacewalk-report channels 
channel_id,channel_label,channel_name,number_of_packages,org_id
123,c7_docker,CentOS 7 Docker-CE,408,1
156,c7_epel,CentOS 7 EPEL,15585,1
125,c7_extras,CentOS 7 Extras,509,1
...
```

If I add `--legacy-report` to 2022.5 the data returns as it did before. (Thanks for the foresight in including this!)

I don't know how to further diagnose this or confirm the hunch that data isn't getting into the new reporting database, so raising as a bug.

Thanks

**Comment by mcalmer** [comment]
Hmm, afaik 2022.05 does not have the new spacewalk-reports tool yet. (@mackdk ?)
But anyway. The tool can only produce data, if the report database was created and filled with data from the taskomatic job.

That job is running 1 time per day in the night. Can you check Admin => Task Schedules => "mgr-update-reporting-bunch" .
Was it running and was it successfully finished?

**Comment by mcalmer** [comment]
Maybe it was 2022.04 which did not had it. Too many version in parallel developed here:-)

**Comment by digdilem** [comment]
Yep - mgt-update-reporting-bunch has failed each stage. 

I did get an email yesterday which I ignored because we've had a lot of taskomatic failure emails since .4 (mentioned in the -users list)
```
Taskomatic bunch mgr-update-reporting-bunch was scheduled to run within the update-reporting-default schedule.

Subtask mgr-update-reporting failed.

For more information check null.
```

Have also `grep mgr-update-reporting-bunch * -R` in /var/log and can't find any error logs relating to this, only http access logs that mention that from me hunting, so unsure how to diagnose where it failed. 

The runtimes for this are interesting. Looks like it's failed after 51 minutes since 2022.5 was upgraded on Sunday. Before then (as .4) it ran for much longer each time.

![image](https://user-images.githubusercontent.com/4052275/168832040-411c8f5f-c3f6-4536-9875-e13df69bfa2b.png)

**Comment by mcalmer** [comment]
Is there something in rhn_taskomatic_daemon.log ?
Maybe you can do a 

```
$> tail -f /var/log/rhn/rhn_taskomatic_daemon.log
```
and then hit the schedule button on that web page. Then you can see if that cause stacktraces.

**Comment by digdilem** [comment]
Certainly.   This is rhn_taskomatic_daemon.log from this morning's run. 

[rhn_taskomatic_daemon.log](https://github.com/uyuni-project/uyuni/files/8714044/rhn_taskomatic_daemon.log)

The gui says this, so I've included all the logs for that period. 

![image](https://user-images.githubusercontent.com/4052275/168981125-9571682a-5402-444b-bdbf-3be8452fb8a9.png)

(15:17 run yesterday was manually triggered as per your advice, 00:00 was as per default schedule)


I also got these failure emails overnight

```
0030

Taskomatic bunch update-payg-data-bunch was scheduled to run within the update-payg-default schedule.

Subtask update-payg-hosts failed.

For more information check null.

0040
Taskomatic bunch update-payg-data-bunch was scheduled to run within the update-payg-default schedule.

Subtask update-payg-hosts finished successfully and is back to normal.
```

Hopefully this is useful

**Comment by mcalmer** [comment]
Here we have the problem:
```
2022-05-18 00:51:55,618 [DefaultQuartzScheduler_Worker-8] ERROR com.redhat.rhn.common.db.datasource.CachedStatement - Error while processing cached statement sql: INSERT INTO XccdScanResult (mgm_id, synced_date, rule_id,rulesystem,result,system_id,ident,scan_id,idref) VALUES (1, current_timestamp, ?,?,?,?,?,?,?)
com.redhat.rhn.common.db.WrappedSQLException: ERROR: duplicate key value violates unique constraint "xccdscanresult_pk"
  Detail: Key (mgm_id, scan_id, rule_id)=(1, 5, 4788) already exists.
```

Now we know were we have to look at. Thanks.

**Comment by digdilem** [comment]
After installing 2022.06 yesterday, I triggered a manual run of `mgr-update-reporting-bunch`  which took around 50 minutes before failing. 

However, it ran at its  scheduled time of midnight and I was pleased to receive an email this morning telling me that it had completed successfully after  a 1 hour run - the first successful run since I upgraded Uyuni to 2022.05 and it failed. 

On checking the spacewalk-report output now, without using --legacy, I'm pleased to report that I appear to have a full set of reports working, so it looks like this fix has worked. 

Thanks for your work.

**Comment by tacerus** [comment]
Hello!

I receive emails like

```
Taskomatic bunch mgr-update-reporting-bunch was scheduled to run within the update-reporting-default schedule.

Subtask mgr-update-reporting failed.

For more information check null.
```

as well - should I open a new issue? Given this is a fresh installation, I don't think there should be anything returning "null" by default.
